kore/src/Kore/Unification/Unify.hs

```diff
@@ -52,43 +67,75 @@ class Monad unifier => MonadUnify unifier where
         -> unifier ()
     explainBottom _ _ _ = pure ()
 
-instance MonadUnify unifier => MonadUnify (BranchT unifier) where
-    throwSubstitutionError = Monad.Trans.lift . throwSubstitutionError
-    {-# INLINE throwSubstitutionError #-}
-
-    throwUnificationError = Monad.Trans.lift . throwUnificationError
-    {-# INLINE throwUnificationError #-}
-
-    liftSimplifier = Monad.Trans.lift . liftSimplifier
-    {-# INLINE liftSimplifier #-}
-
-    explainBottom why term1 term2 =
-        Monad.Trans.lift (explainBottom why term1 term2)
-    {-# INLINE explainBottom #-}
+-- | 'UnifierT' contains everything that is needed for a MonadUnify,
+-- but allows parameterization over a monad transformer.
+-- See also: 'Unifier'.
+newtype UnifierT m a = UnifierT
```

Comment:
Please give this an explicit kind signature
```suggestion
newtype UnifierT (t :: (* -> *) -> * -> *) a = UnifierT
```
I found it quite confusing that this is a transformer-transformer rather than a transformer!

Reply:
Should we also rename it to `UnifierTT`?